dolfyn/velocity.py

```diff
@@ -0,0 +1,543 @@
+    ========
+    :class:`VelBinner`
+
+    """
+    def __init__(self, ds, *args, **kwargs):
+        self.ds = ds
+
+    @property
+    def u(self,):
+        """The first velocity component.
+
+        This is simply a shortcut to self['vel'][0]. Therefore,
+        depending on the coordinate system of the data object
+        (self.attrs['coord_sys']), it is:
+
+        - beam:      beam1
+        - inst:      x
+        - earth:     east
+        - principal: streamwise
+        """
+        return self.ds['vel'][0]
+
+    @property
+    def v(self,):
+        """The second velocity component.
+
+        This is simply a shortcut to self['vel'][1]. Therefore,
+        depending on the coordinate system of the data object
+        (self.attrs['coord_sys']), it is:
+
+        - beam:      beam2
+        - inst:      y
+        - earth:     north
+        - principal: cross-stream
+        """
+        return self.ds['vel'][1]
+
+    @property
+    def w(self,):
+        """The third velocity component.
+
+        This is simply a shortcut to self['vel'][2]. Therefore,
+        depending on the coordinate system of the data object
+        (self.attrs['coord_sys']), it is:
+
+        - beam:      beam3
+        - inst:      z
+        - earth:     up
+        - principal: up
+        """
+        return self.ds['vel'][2]
+
+    @property
+    def U(self,):
+        """Horizontal velocity as a complex quantity
+        """
+        return xr.DataArray(
+                    (self.u + self.v * 1j),
+                    attrs={'units':'m/s',
+                           'description':'horizontal velocity (complex)'})
+    @property
+    def U_mag(self,):
+        """Horizontal velocity magnitude
+        """
+        return xr.DataArray(
+                    np.abs(self.U),
+                    attrs={'units':'m/s',
+                           'description':'horizontal velocity magnitude'})                            
+    @property
+    def U_dir(self,):
+        """Angle of horizontal velocity vector, deg clockwise from
+        X/North/streamwise. Direction is 'to', as opposed to 'from'.
+        """
+        # Convert from radians to degrees
+        angle = np.angle(self.U)*(180/np.pi)
+        
+        if hasattr(self.ds, 'coord_sys') and self.ds.coord_sys=='earth':
+            # Convert heading from East to North
+            angle -= 90 # -90 degrees East == 0 degrees North
+            idn = np.where(angle<-179)
+            idp = np.where(angle>180)
+            angle[idn] += 360
+            angle[idp] -= 360
+            dr = 'North'
+        else:
+            dr = 'East'
+        
+        return xr.DataArray(angle,
+                            dims=self.U.dims,
+                            coords=self.U.coords,
+                            attrs={'units':'deg',
+                           'description':'horizontal velocity flow direction, CW from X/' 
+                                          +dr+'/streamwise'})
+    @property
```

Comment:
I think we need to drop this. It's not well-constrained enough to be defined here.

Reply:
I can see that. Thought it was weird to define direction as clockwise from east